tools/testtool/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"dagger.io/dagger"
+)
+
+func main() {
+	sandbox := flag.Bool("sandbox", false, "Run the provided command in a sandboxed environment instead of your native environment")
+	flag.Parse()
+	if flag.NArg() < 2 {
+		log.Fatalln("too few arguments")
+	}
+
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stdout))
+	if err != nil {
+		log.Fatalf("unable to connect to client: %v\n", err)
+	}
+	defer client.Close()
+
+	var kernel *dagger.File
+	var kpath, qemuCmd string
+
+	switch envWithDefault("UROOT_TESTARCH", runtime.GOARCH) {
+	case "amd64":
+		kpath = "/bzImage"
+		qemuCmd = "qemu-system-x86_64 -L %s -m 1G"
+		kernel = client.Container().From("ghcr.io/hugelgupf/vmtest/kernel-amd64:main").File(kpath)
+	case "arm":
+		kpath = "/zImage"
+		qemuCmd = "qemu-system-aarch64 -machine virt -cpu max -m 1G -L %s"
+		kernel = client.Container().From("ghcr.io/hugelgupf/vmtest/kernel-arm:main").File(kpath)
+	case "arm64":
+		kpath = "/Image"
+		qemuCmd = "qemu-system-arm -M virt,highmem=off -L %s"
+		kernel = client.Container().From("ghcr.io/hugelgupf/vmtest/kernel-arm64:main").File(kpath)
+	}
+
+	artifacts := client.
+		Container().
+		From("ghcr.io/hugelgupf/vmtest/qemu:main").
+		WithFile(kpath, kernel).
+		Directory("/")
+
+	if *sandbox {
+		runSandboxed(ctx, client, artifacts, kpath, qemuCmd, flag.Args())
+		return
+	}
+
+	runNatively(ctx, artifacts, kpath, qemuCmd, flag.Args())
+}
+
+func envWithDefault(key, def string) string {
+	if env, ok := os.LookupEnv(key); ok {
+		return env
+	}
+
+	return def
+}
+
+func runNatively(ctx context.Context, artifacts *dagger.Directory, kpath, qemuCmd string, args []string) {
+	tmp, err := os.MkdirTemp(".", "ci-testing")
+	if err != nil {
+		log.Fatalf("unable to create tmp dir: %v\n", err)
+	}
+	defer os.RemoveAll(tmp)
+
+	if ok, err := artifacts.
+		Export(ctx, tmp); !ok || err != nil {
```

Comment:
This is a case of indentation confusion. It should be unambiguous to a reader whether `Export(ctx, tmp)` is part of the if-statement or the code block. 

https://google.github.io/styleguide/go/decisions#conditional-formatting
https://google.github.io/styleguide/go/decisions#conditionals-and-loops

Do not line break in if-conditions

Reply:
Ah you're right, this was left over from modifying the code in the iterations and initially there were more statements over the export method which would have made the line too long.